### PR TITLE
Initial smart contract docs

### PR DESCRIPTION
**Pull request by ulope**
First attempt at documenting the deployment, usage, and verification of the smart contracts.
Architecture remains TODO.

I'm not really happy with the verification section but for now, have no better idea how to structure it.

### CODE REVIEW

File: docs/smart_contracts.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
+Raiden Network Smart Contracts
+==============================
+
+Smart Contract Architecture
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: The architecture could be part of the spec itself so perhaps not include them (at least for now) in their own section here?

File: docs/smart_contracts.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
+Raiden Network Smart Contracts
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: Related to my previous comment: Perhaps rename this to Smart Contract Deployment

File: docs/smart_contracts.rst [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+==============================
+
+Smart Contract Architecture
+---------------------------
+
+TODO
+
+
+Deployment and Use of the Smart Contracts
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: `Use` -> `Usage`

File: docs/smart_contracts.rst [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+-----------------------------------------
+
+Deploying
+~~~~~~~~~
+
+.. highlight:: bash
+
+To deploy the smart contracts use the provided utility script in
+`tools/deploy.py`. It expects an unencrypted hex-encoded private key as it's
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: `it's` -> `its`

File: docs/smart_contracts.rst [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+#. Modify the URL to :code:`verifyContract2` to use the new verification tool
+#. Enter the contract name (:code:`Registry`) and select the used solc version
+#. Paste the combined contract code into the provided text area
+   The combined contract code can be generated with the
+   :code:`tools/join-contracts.py` utility script::
+
+    cd raiden/smart_contracts
+    python ../../tools/join-contracts.py Registry.sol registry-joined.sol
+
+#. Enter :code:`ChannelManagerLibrary` and it's corresponding address into the
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: Add a bullet point specifying `Optimized` or not and to say that by default we don't optimize at the moment

Comment (LefterisJP) [inline docs/smart_contracts.rst]: `it's` -> `its`

File: docs/smart_contracts.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+   contract
+#. In this case this is just the token address. To get the right format you can
+   use pyethereum's :code:`encode_abi` function. For example::
+
+    >>> from ethereum.abi import encode_abi
+    >>> encode_abi(['address'], ['0f114a1e9db192502e7856309cc899952b3db1ed']).encode('hex')
+    '0000000000000000000000000f114a1e9db192502e7856309cc899952b3db1ed'
+
+#. Place the output from the above call into the "Constructor Arguments" field
+#. Enter :code:`ChannelManagerLibrary` and it's corresponding address into the
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: `it's` -> `its`

File: docs/smart_contracts.rst [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+    >>> from ethereum.abi import encode_abi
+    >>> encode_abi(
+    ...     ['address', 'address', 'address', 'uint256'],
+    ...     ['0f114a1e9db192502e7856309cc899952b3db1ed', '001ee1b9b78de26879ac9db3854ce1430b339bee', '2222222222222222222222222222222222222222', 90]
+    ... ).encode('hex')
+    0000000000000000000000000f114a1e9db192502e7856309cc899952b3db1ed000000000000000000000000001ee1b9b78de26879ac9db3854ce1430b339bee0000000000000000000000002222222222222222222222222222222222222222000000000000000000000000000000000000000000000000000000000000005a
+
+#. Place the output from the above call into the "Constructor Arguments" field
+#. Enter :code:`NettingChannelLibrary` and it's corresponding address into the
```
Comment (LefterisJP) [inline docs/smart_contracts.rst]: `it's` -> `its`

File: docs/smart_contracts.rst (hunk 1 above)
Comment (ulope) [reply docs/smart_contracts.rst]: Ok, if you think it fit's better over there I'll change the heading.

File: docs/smart_contracts.rst (hunk 7 above)
Comment (ulope) [reply docs/smart_contracts.rst]: I'm seeing a theme here ;)

**Pull request merged**